### PR TITLE
Custom homepage

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves : https://github.com/brave/brave-browser/issues/8217

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/ntp_background_images/android/ntp_background_images_bridge.cc [hunk 1]
```diff
@@ -25,6 +25,7 @@
 #include "components/keyed_service/content/browser_context_dependency_manager.h"
 #include "content/public/browser/browser_context.h"
 #include "content/public/browser/browser_thread.h"
+#include "brave/components/ntp_background_images/browser/ntp_background_images_data.h"
```
Comment (simonhong) [inline browser/ntp_background_images/android/ntp_background_images_bridge.cc:28]: nit: alphabet order including.

File: android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageView.java [hunk 2]
```diff
@@ -242,16 +264,39 @@ else if (seconds > 60) {
...
         NTPUtil.updateOrientedUI(mTabImpl.getActivity(), mNewTabPageLayout);
-
-        if(BravePrefServiceBridge.getInstance().getBoolean(BravePref.NTP_SHOW_BACKGROUND_IMAGE)
+        ImageView mSponsoredLogo = (ImageView)mNewTabPageLayout.findViewById(R.id.sponsored_logo);
+        FloatingActionButton mSuperReferralLogo = (FloatingActionButton) mNewTabPageLayout.findViewById(R.id.super_referral_logo);
+        if (ntpImage instanceof Wallpaper 
+                && isReferralEnabled()) {
+            setBackgroundImage(ntpImage);
+            Log.e("NTP", "Theme name : "+ mNTPBackgroundImagesBridge.getSuperReferralThemeName());
+            Log.e("NTP", "Is Super Referral : "+ mNTPBackgroundImagesBridge.isSuperReferral());
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageView.java:280]: I don't think we need this logs

File: android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageView.java [hunk 3]
```diff
@@ -352,26 +399,146 @@ public void updateNTPImage() {
...
+            layoutParams.weight = 0.25f;
+            layoutParams.gravity = Gravity.CENTER;
+            view.setLayoutParams(layoutParams);
+            view.setOnCreateContextMenuListener(new View.OnCreateContextMenuListener() {
+                @Override
+                public void onCreateContextMenu(ContextMenu menu, View v, ContextMenu.ContextMenuInfo menuInfo) {
+                    menu.add(R.string.contextmenu_open_in_new_tab).setOnMenuItemClickListener(new MenuItem.OnMenuItemClickListener() {
+                        @Override
+                        public boolean onMenuItemClick(MenuItem item) {
+                            Log.e("NTP", "normal tab");
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageView.java:500]: logs

File: android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageView.java [hunk 4]
```diff
@@ -352,26 +399,146 @@ public void updateNTPImage() {
...
+                            openNewTab(false, topSite.getDestinationUrl());
+                            return true;
+                        }
+                    });
+                    menu.add(R.string.contextmenu_open_in_incognito_tab).setOnMenuItemClickListener(new MenuItem.OnMenuItemClickListener() {
+                        @Override
+                        public boolean onMenuItemClick(MenuItem item) {
+                            Log.e("NTP", "incognito");
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageView.java:508]: logs

File: android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageView.java [hunk 5]
```diff
@@ -352,26 +399,146 @@ public void updateNTPImage() {
...
+                            openNewTab(true, topSite.getDestinationUrl());
+                            return true;
+                        }
+                    });
+                    menu.add(R.string.contextmenu_save_link).setOnMenuItemClickListener(new MenuItem.OnMenuItemClickListener() {
+                        @Override
+                        public boolean onMenuItemClick(MenuItem item) {
+                            Log.e("NTP", "save");
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageView.java:516]: logs

File: android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageView.java [hunk 6]
```diff
@@ -352,26 +399,146 @@ public void updateNTPImage() {
...
+                            RequestCoordinatorBridge.getForProfile(mProfile).savePageLater(
+                                topSite.getDestinationUrl(), OfflinePageBridge.NTP_SUGGESTIONS_NAMESPACE, true /* userRequested */);
+                            return true;
+                        }
+                    });
+                    menu.add(R.string.remove).setOnMenuItemClickListener(new MenuItem.OnMenuItemClickListener() {
+                        @Override
+                        public boolean onMenuItemClick(MenuItem item) {
+                            Log.e("NTP", "remove");
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageView.java:525]: logs

File: android/java/org/chromium/chrome/browser/ntp_background_images/util/NTPUtil.java [hunk 7]
```diff
@@ -344,6 +352,56 @@ public static Bitmap getWallpaperBitmap(NTPImage ntpImage, int layoutWidth, int
         return bitmapWithGradient;
     }
 
+    public static Bitmap getTopSiteBitmap(String iconPath) {
+        Context mContext = ContextUtils.getApplicationContext();
+        InputStream inputStream = null;
+        Bitmap topSiteIcon = null;
+        try {
+            Uri imageFileUri = Uri.parse("file://"+ iconPath);
+            inputStream = mContext.getContentResolver().openInputStream(imageFileUri);
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/ntp_background_images/util/NTPUtil.java:361]: do we execute that on a correct thread? I mean don't we block UI thread on that place and have a StrictMode warning?

Comment (deeppandya) [reply android/java/org/chromium/chrome/browser/ntp_background_images/util/NTPUtil.java:361]: @SergeyZhukovsky this function is called from a separate thread from BraveNewTabPageView.

**Pull request merged**